api/lib/opentelemetry/distributed_context/propagation/http_text_format.rb

```diff
@@ -17,7 +16,22 @@ module Propagation
       # Propagation is usually implemented via library-specific request interceptors, where the client-side injects values
       # and the server-side extracts them.
       class HTTPTextFormat
-        # TODO
+        def extract(carrier, &block)
```

Comment:
Would an implicit block be better since we opt for yielding and not using the explicit `&block` param.

In either case can we document the parameters to this method as well?

Reply:
imo with an implicit block you have to read the documentation/implementation rather than the function signature.   If you feel strongly I don't mind.

documentation for sure.

Reply:
I did the changes, have a look.  Not so sure about my documentation though.

Reply:
Looks good. The alternative would have been to use the block param (block.call) instead of using `yield`, but its slightly faster to use yield with an implicit block.

Reply:
So I looked through the rails source, and I think a good rule is, only name the block if we pass it down to another method.  Then it's clear the yield is getting "passed" as it were.